src/javascript/core/store.js

```diff
@@ -116,14 +116,20 @@ const Store = function (name, config = {}) {
 	const oldCookieStoreData = cookieLoad(this.storageKey);
 	if (oldCookieStoreData) {
 		try {
-			const data = JSON.parse(oldCookieStoreData);
-			if (this.data) {
-				Object.assign(this.data, data);
+			if (this.storageKey === 'spoor-id') {
+				// spoor-id is stored directly as a string and not as an object
+				this.data = oldCookieStoreData;
+				cookieRemove('spoor-id');
```

Comment:
I can see the cookie removal being an issue for id consistency.
Scenario: user was on ft.com and has cookie set. user is put in the beta homepage cohort. local storage is set, cookie is removed. user clicks on an article from the beta homepage: a new cookie is set, creating another id mismatch.

Reply:
oh wow, yes, great catch Lily!